test/Geometry/axistensor_conversion_benchmarks.jl

```diff
@@ -0,0 +1,102 @@
+using Test
+using StaticArrays
+import Combinatorics
+import OrderedCollections
+import Random
+#! format: off
+using ClimaCore.Geometry:
+    AbstractAxis,
+    CovariantAxis,
+    AxisVector,
+    ContravariantAxis,
+    LocalAxis,
+    CartesianAxis,
+    AxisTensor,
+    Axis2Tensor
+
+include("transform_project.jl") # compact, generic but unoptimized reference
+
+function all_axes()
+    all_Is() = [(1,), (2,), (3,), (1, 2), (1, 3), (2, 3), (1, 2, 3)]
+    collect(Iterators.flatten(map(all_Is()) do I
+        (
+            CovariantAxis{I}(),
+            ContravariantAxis{I}(),
+            LocalAxis{I}(),
+            CartesianAxis{I}()
+        )
+    end))
+end
+
+function all_axistensors(::Type{FT}) where {FT}
+    axis_vecs = map(all_axes()) do ax # N = 1
+        sv = @SVector rand(FT, 3)
+        AxisTensor((ax,), sv)
+    end
+    axis_tensors = map(Combinatorics.combinations(all_axes(), 2)) do ax_combo
+        sm = @SMatrix rand(FT, 3,3)
+        AxisTensor(Tuple(ax_combo), sm)
+    end
+    return vcat(axis_vecs, axis_tensors)
+end
+
+function get_conversion_args(::Type{FT}, fun::F) where {FT, F}
+    aat = all_axistensors(FT)
+    aat = aat[1:10] # length is 406
```

Comment:
cc @simonbyrne, I've set this to sub-sample for now, because testing all of them takes ~5 min.